e3nn/nn/normact.py

```diff
@@ -0,0 +1,87 @@
+import torch
+
+from e3nn.o3 import ElementwiseTensorProduct, Norm
+
+class NormActivation(torch.nn.Module):
+    r"""Gate activation function
+
+    Parameters
+    ----------
+    irreps_scalars : `Irreps`
+        representation of the scalars (the one not use for the gates)
+
+    act_scalars : list of function or None
+        activations acting on the scalars
+
+    irreps_gates : `Irreps`
+        representation of the scalars used to gate.
+        ``irreps_gates.num_irreps == irreps_nonscalars.num_irreps``
+
+    act_gates : list of function or None
+        activations acting on the gates
+
+    irreps_nonscalars : `Irreps`
+        representation of the non-scalars.
+        ``irreps_gates.num_irreps == irreps_nonscalars.num_irreps``
+
+    Examples
+    --------
+
+    >>> g = Gate("16x0o", [torch.tanh], "32x0o", [torch.tanh], "16x1e+16x1o")
+    >>> g.irreps_out
+    16x0o+16x1o+16x1e
+    """
+    def __init__(self,
+                 irreps_in,
+                 scalar_nonlinearity,
+                 normalize=True,
+                 bias=False,
+                 epsilon=1e-8,
+                 **kwargs):
+        super().__init__()
+        self.irreps_in = irreps_in
+        self.irreps_out = irreps_in
```

Comment:
I'm use to call `o3.Irreps(irreps_in)` in order to allow the user to do `NormActivation("0e + 1e")`